source/agora/test/Base.d

```diff
@@ -248,11 +249,19 @@ private UnitTestResult customModuleUnitTester ()
             CircularAppender!()().print(output);
             stdout.flush();
         }
+        return false;
     }
 
     // Run single-threaded tests
+    bool failed_single_threaded;
     foreach (mod; single_threaded)
```

Comment:
I wonder if 'run in single threaded' and 'fail at first error' should be 2 separate aspect/env variables. The reason is that for local development I would still like to 'run single threaded', but not 'fail at first error'. If 4 tests fails, then I would just rerun those 4 one-by-one. (Sometimes tests fails even in single threaded on my machine, even if there are no bugs in the code, so I just generally rerun them one-by-one)

Reply:
How does it look now ?